source/posix/socket.c

```diff
@@ -646,14 +635,17 @@ int aws_socket_connect(
     AWS_ZERO_STRUCT(address);
     socklen_t sock_size = 0;
     int pton_err = 1;
+    bool port_is_bad = false;
```

Comment:
This name is too generic. Maybe `port_in_range_for_socket_type`?

Reply:
there's a separate port-validation function now